src/map/script.c

```diff
@@ -8841,12 +8867,13 @@ static BUILDIN(makeitem2)
 	}
 
 	if (m == -1) {
-		ShowError("makeitem2: Nonexistant map requested.\n");
+		ShowError("buildin_makeitem2: invalid map name.\n");
 		return true;
 	}
 
-	x = (script_hasdata(st, 12) ? script_getnum(st, 12) : 0);
-	y = (script_hasdata(st, 13) ? script_getnum(st, 13) : 0);
+	int16 x = (script_hasdata(st, 12) ? script_getnum(st, 12) : 0);
+	int16 y = (script_hasdata(st, 13) ? script_getnum(st, 13) : 0);
+	int16 range;
```

Comment:
It was already this way before your changes, but please change these to int, for the same reasoning as above (and the scope of `range` can be reduced and moved to the appropriate else branch, to avoid keeping this uninitialized variable around)

```suggestion
	int x = (script_hasdata(st, 12) ? script_getnum(st, 12) : 0);
	int y = (script_hasdata(st, 13) ? script_getnum(st, 13) : 0);
```